gateway/bucketfs.go

```diff
@@ -78,12 +78,12 @@ func serveBucket(fs serveBucketFileSystem) gin.HandlerFunc {
 	}
 }
 
-func (f *bucketFileSystem) GetThread(ctx context.Context, key string) (id thread.ID, err error) {
-	ik, err := f.keys.Get(ctx, key)
+func (f *bucketFileSystem) GetThread(ctx context.Context, bkey string) (id thread.ID, err error) {
+	key, err := f.keys.GetByCid(ctx, bkey)
```

Comment:
Very much the same as before, but the key is indexed by name (for the core IPFS API), but we want CID here.